### PR TITLE
Fixed #35306 -- Improved documentation for the various date format settings

### PR DESCRIPTION
**Pull request by lufafajoshua**
# Trac ticket number

ticket-35306

# Branch description
Updated documentation on localization formats usage and precedence.

# Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" **ticket flag** in the Trac system.
- [ ] I have added or updated relevant **tests**.
- [x] I have added or updated relevant **docs**, including release notes if applicable.
- [ ] For UI changes, I have attached **screenshots** in both light and dark modes.

### CODE REVIEW

**Comment by nessita** [comment]
@lufafajoshua thank you for your work.

When this PR is ready for review, please update the checklist from the PR description, and don't forget to set the proper ticket flags for this to be listed in the branches needing review.

File: docs/topics/i18n/formatting.txt [hunk 1]
```diff
@@ -60,6 +60,9 @@ to be machine-readable, you will always want unlocalized values. You
 may also want to use localization in selected templates, rather than
 using localization everywhere.
 
+The format used is the one from the settings and under normal circumstances
+the locale dictated format has a higher precedence and will be applied instead.
```
Comment (claudep) [inline docs/topics/i18n/formatting.txt:64]: What format? Which settings? Sorry, but I don't think this adds much clarity to these docs.

Maybe something like (once again I'm not a skilled English writer):
`When localization is disabled through one of the tag or filter described below, the localization formats used are taken from the general formatting settings (list of settings here?).`

**Review by claudep — CHANGES_REQUESTED** [review]

File: docs/topics/i18n/formatting.txt (hunk 1 above)
Comment (lufafajoshua) [reply docs/topics/i18n/formatting.txt:64]: Hey @claudep , thanks for your review and response on this, Just for clarity, is it okay to pass a reference to the [globalization settings](https://docs.djangoproject.com/en/5.0/ref/settings/#globalization-i18n-l10n)  as the general formatting settings instead of listing them.

Comment (claudep) [reply docs/topics/i18n/formatting.txt:64]: Good idea for this link! Now that I see that list, maybe it would be nice to distinguish in that list the settings that may overriding by locale formats, and those that don't, and then suggest (maybe in a different PR) separating that list in two sub-lists. Then we would have a perfect section to link to in this PR.

File: docs/ref/settings.txt [hunk 2]
```diff
@@ -1838,21 +1838,23 @@ A list of IP addresses, as strings, that:
...
-A string representing the language code for this installation. This should be in
-standard :term:`language ID format <language code>`. For example, U.S. English
-is ``"en-us"``. See also the `list of language identifiers`_ and
-:doc:`/topics/i18n/index`.
-
-:setting:`USE_I18N` must be active for this setting to have any effect.
+A string representing the language code for this installation.
+This should be in standard :term:`language ID format <language code>`.
+For example, U.S. English is ``"en-us"``. See also the
+`list of language identifiers`_ and :doc:`/topics/i18n/index`.
```
Comment (sarahboyce) [inline docs/ref/settings.txt:1844]: This is the same, can you revert these changes?

File: docs/ref/settings.txt [hunk 3]
```diff
@@ -1838,21 +1838,23 @@ A list of IP addresses, as strings, that:
...
 * If the locale middleware isn't in use, it decides which translation is served
   to all users.
 * If the locale middleware is active, it provides a fallback language in case the
   user's preferred language can't be determined or is not supported by the
   website. It also provides the fallback translation when a translation for a
   given literal doesn't exist for the user's preferred language.
+* If localization is explicitly disabled via the :tfilter:`localize` and
+  :tfilter:`unlocalize` template filters or tags, it provides fallback
+  localization formats which will be applied instead. see
+  :ref:`controlling localization in templates <topic-l10n-templates>`.
```
Comment (sarahboyce) [inline docs/ref/settings.txt:1857]: I think as we're referencing explicitly disabling localization, we should reference 
```
:tfilter:`unlocalize`
```
and 
```
:ttag:`{% localize off %} <localize>`
```
I think it might be wrong to reference
```
:tfilter:`localize`
```
when talking about disabling localization

File: docs/ref/settings.txt [hunk 4]
```diff
@@ -1838,21 +1838,22 @@ A list of IP addresses, as strings, that:
...
 
 * If the locale middleware isn't in use, it decides which translation is served
   to all users.
 * If the locale middleware is active, it provides a fallback language in case the
   user's preferred language can't be determined or is not supported by the
   website. It also provides the fallback translation when a translation for a
   given literal doesn't exist for the user's preferred language.
+* If localization is explicitly disabled via the :tfilter:`localize` and
+  :tfilter:`unlocalize` template filters or tags. see
+  :ref:`controlling localization in templates <topic-l10n-templates>`.
```
[the remark this answers is not in the document]
Comment (sarahboyce) [reply docs/ref/settings.txt:1858]: Yes, "see" needs to be capitalized and it would be good to end with "for [insert something]".  
You can just say "for more details" but you might want to be more specific like "for further explanation on [insert something]."
Rough example:
```
See :ref:`controlling localization in templates <topic-l10n-templates>` for more details.
```

File: docs/topics/i18n/formatting.txt (hunk 1 above)
Comment (sarahboyce) [reply docs/topics/i18n/formatting.txt]: It sounds like we still want to do this, it can be a separate commit or another PR 👍

File: docs/ref/settings.txt [hunk 5]
```diff
@@ -3666,6 +3668,9 @@ Forms
 * :setting:`FORM_RENDERER`
 * :setting:`FORMS_URLFIELD_ASSUME_HTTPS`
 
+
+.. _globalization-settings:
```
Comment (sarahboyce) [inline docs/ref/settings.txt:3672]: This is a new unused reference, I believe the start of implementing [this comment](https://github.com/django/django/pull/18028#discussion_r1567427192).

**Review by sarahboyce — COMMENTED** [review]
Thank you for this @lufafajoshua ⭐ I have a few comments
Can you also update the PR title to better represent what this PR is trying to acheive?

**Review by sarahboyce — APPROVED** [review]
Thank you @lufafajoshua 👍 this looks good to me!
I have squashed commits and applied minor updates, want to check other people are happy before merging but I think we're very close

**Review by claudep — APPROVED** [review]
LGTM, thanks!
